### ISSUE TITLE
multi GPU training problem

### ISSUE THREAD

**Issue by MiSsU-HH** [VSainteuf/utae-paps#9]
When I use multiple GPUs for training, the inconsistent size of tensor types in instance_masks leads to data parallelism errors. How can I solve this error?

**Comment by VSainteuf** [comment]
So instance_masks is just a list of tensors. The length is equal to the number of centers that are detected in a given sample (the local maxima of centerness), hence the variability. So one possible workaround would be to pad this list to a larger constant length.

One way to do this is to replace l.215 of `src/panoptic/paps.py` with:
`instance_masks = [torch.zeros(1, device=stack.device) for _ in range(10000)]`
and l.241 with :
`instance_masks[i]=pred_mask`

**Comment by MiSsU-HH** [comment]
I think it may be caused by different sizes of tensors in instance_mask:
![image](https://user-images.githubusercontent.com/81288897/165749969-27b00b6b-d57c-461d-a117-322824ba7c71.png)

**Comment by VSainteuf** [comment]
Ah ok I see. Can you make sure that this is the problem ? If you replace the tensors in instance_mask with a list of dummy tensors with constant shape, does code run on multi-gpu? Also, did you make sure that the code runs properly on a single gpu?

If this is really the problem, I don't think there is a simple workaround. One option would be to store the candidate instance masks in a tensor of shape NxHxW with as many channels as there are candidate instances. And maybe use sparse tensors to avoid useless memory usage. But that would involve a bit more changes to the code.

**Comment by MiSsU-HH** [comment]
I make sure that the code runs properly on a single gpu. And I replace the tensors in instance_mask with a list of dummy tensors with constant shape, the code runs properly on multi gpu.

**Comment by MiSsU-HH** [comment]
Do you have other solutions to complete multi GPU training?

**Comment by MiSsU-HH** [comment]
Hi, I have solved this problem by replacing DataParallel model with DistributedDataParallel.
